docs/eBpfExtensions.md

```diff
@@ -296,14 +296,73 @@ The helper function ID for a general helper function must be in the range 0 - 65
 The parameter and return types for these helper functions must adhere to the `ebpf_argument_type_t` and
 `ebpf_return_type_t` enums.
 
-### 2.7 Registering Program Types and Attach Types
+### 2.7 Registering Program Types and Attach Types - eBPF Store
 The eBPF Execution Context loads an eBPF program from an ELF file that has program section(s) with section names. The
 prefix to these names determines the program type. For example, the section name `"xdp"` implies that the corresponding
-program type is `EBPF_PROGRAM_TYPE_XDP`. The Execution Context discovers the program type associated with a section
-prefix by reading the data from Windows registry. When an eBPF extension is installed, it must update the registry with
-the program types it implements along with the associated section prefixes.
-
-_Note: The registry location and data format are TBD. This is currently tracked by issue #223._
+program type is `EBPF_PROGRAM_TYPE_XDP`.
+
+The *Execution Context* discovers the program type associated with a section prefix by reading the data from Windows registry,
+in what is referred to aa the ***"eBPF Store"***.
+When an eBPF extension is installed, it must update the eBPF store with the program types it implements along with the associated section prefixes.
+
+To operate on the eBPF store, the extension must link the `\build\native\lib\ebpf_store_helper_km.lib` kernel-mode library and include the related `\build\native\include\ebpf_store_helper.h` header file, both distributed within the [eBPF for Windows NuGet package](https://www.nuget.org/packages/eBPF-for-Windows/). With these, the extension can use the following APIs to register program types, attach types and helper functions:
+
+- `ebpf_store_update_helper_prototype`: updates the provider prototype information in the eBPF registry store, given a pointer to the registry key to be initialized and a pointer to the helper function prototype (i.e., `_ebpf_helper_function_prototype`):
```

Comment:
```suggestion
- `ebpf_store_update_helper_prototype`: updates the provider prototype information in the eBPF store, given a pointer to the registry key to be initialized and a pointer to the helper function prototype (i.e., `_ebpf_helper_function_prototype`):
```
Also, I'm confused about one part... the point of the "ebpf store" helper is to abstract away whether it's stored in the registry or the filesystem or whatever else.  But "given a pointer to the registry key" does not seem consistent with that.  The "ebpf store" helper API should not mention registry.